base/db/R/get.trait.data.R

```diff
@@ -309,6 +406,38 @@ get.trait.data <- function(pfts, modeltype, dbfiles, database, forceupdate, trai
   invisible(result)
 }
 
+#' Symmetric set difference of two data frames
+#'
+#' @param x,y `data.frame`s to compare
+#' @param xname Label for data in x but not y. Default = "x"
+#' @param yname Label for data in y but not x. Default = "y"
+#' @param namecol Name of label column. Default = "source".
+#' @return `data.frame` of data not common to x and y, with additional
+#'   column (`namecol`) indicating whether data are only in x
+#'   (`xname`) or y (`yname`)
+#' @export
+#' @examples
+#' xdf <- data.frame(a = c("a", "b", "c"),
+#'                   b = c(1, 2, 3),
+#'                   stringsAsFactors = FALSE)
+#' ydf <- data.frame(a = c("a", "b", "d"),
+#'                   b = c(1, 2.5, 3),
+#'                   stringsAsFactors = FALSE)
+#' symmetric_setdiff(xdf, ydf)
+symmetric_setdiff <- function(x, y, xname = "x", yname = "y",
+                              namecol = "source") {
+  stopifnot(is.data.frame(x), is.data.frame(y),
+            is.character(xname), is.character(yname),
+            length(xname) == 1, length(yname) == 1)
+  namecol <- dplyr::sym(namecol)
+  xy <- dplyr::setdiff(x, y) %>%
+    dplyr::mutate(!!namecol := xname)
+  yx <- dplyr::setdiff(y, x) %>%
+    dplyr::mutate(!!namecol := yname)
+  dplyr::bind_rows(xy, yx) %>%
+    dplyr::select(!!namecol, dplyr::everything())
+}
+
```

Comment:
Placeholder to say this is as far as I got in my first reviewing pass. Will try to look at the rest over the weekend.